### DISCUSSION TITLE
Ignore warnings in generated files

### DISCUSSION THREAD

**Discussion by ryan-morris** [mganss/XmlSchemaClassGenerator#525] [General]
We have a lot of analyzers that the generated files are triggering. Is it possible to add something like `#pragma warning disable
` into the generated files? I tried doing TypeVisitor and adding annotations per class but some of our analyzers run on file level, for example specific file header with stylecop etc.

**Comment by mganss** [comment]
One option that comes to mind is to create your own `OutputWriter` that adds these pragmas.

Another option would be to configure StyleCop to exclude generated code using the `excludeGeneratedCode` property.
